lib/Serialization/Deserialization.cpp

```diff
@@ -4428,7 +4457,7 @@ Expected<Type> ModuleFile::getTypeChecked(TypeID TID) {
     decls_block::UnboundGenericTypeLayout::readRecord(scratch,
                                                       genericID, parentID);
 
-    auto genericDecl = cast<NominalTypeDecl>(getDecl(genericID));
+    auto genericDecl = cast<GenericTypeDecl>(getDecl(genericID));
```

Comment:
Thanks! I couldn't think of a test case because I didn't know where UnboundGenericTypes actually got serialized. I'll split it out into its own commit and use your test case.